addons/mrp/mrp.py

```diff
@@ -151,6 +151,10 @@ class mrp_bom(osv.osv):
     _description = 'Bill of Material'
     _inherit = ['mail.thread']
 
+    def _get_bom_ids(self, cr, uid, product_template_ids, context=None):
+        # It will return BOM ids of product.
+        return self.pool['mrp.bom'].search(cr, uid, [('product_tmpl_id', 'in', product_template_ids)], context=context)
```

Comment:
this is a trigger function. it is called from product.template object. so self is product.template, not mrp.bom.